swift/ql/lib/codeql/swift/elements/Element.qll

```diff
@@ -1,18 +1,7 @@
 private import codeql.swift.generated.Element
 
 class Element extends Generated::Element {
-  private predicate resolvesFrom(Element e) { e.getResolveStep() = this }
-
   override string toString() { result = this.getPrimaryQlClasses() }
-
-  Element getFullyUnresolved() {
-    not this.resolvesFrom(_) and result = this
-    or
-    exists(Element e |
-      this.resolvesFrom(e) and
-      result = e.getFullyUnresolved()
-    )
-  }
```

Comment:
Is it possible to have a hideable parent of a non-hideable child?

Reply:
you mean in AST parent/child terms, or in superclass/subclass terms? Both can mix, which just made me think of an issue with this change: hideability should actually travel both directions of inheritance. For example marking `Expr` explicitly as hideable, doesn't only mean that all `Expr` subclasses are hideable, but also superclasses such as `AstNode`. Failing to do so like I'm doing here means that something like `BraceStmt.getElement/1` would fail to hide conversions which would be quite confusing.

Reply:
The worry that prompted this question was the removal of `getFullyUnresolved()` calls from non-hideable elements.  Which seems maybe wrong if a hideable element can resolve to it (I'm not really sure why we use `getFullyUnresolved()` though).